### PR TITLE
BugFix: fix #5309: Be upgrade failed

### LINKED ISSUE #5309: Upgrade Be failed from starrocks-1.19.5 to starrocks-2.0.1
### Steps to reproduce the behavior (Required)
1. create a cluster use apache-doris 0.13
2. upgrade cluster to starrocks-1.19.x
3. upgrade cluster to starrocks-2.0.x

### Expected behavior (Required)
upgrade success
### Real behavior (Required)
Start be failed

*** Aborted at 1650262403 (unix time) try “date -d @1650262403” if you are using GNU date ***
PC: @ 0x7f216b5e2387 __GI_raise
*** SIGABRT (@0x10104) received by PID 65796 (TID 0x7f21661f7700) from PID 65796; stack trace: ***
@ 0x33a1be2 google::(anonymous namespace)::FailureSignalHandler()
@ 0x7f216c2ad630 (unknown)
@ 0x7f216b5e2387 __GI_raise
@ 0x7f216b5e3a78 __GI_abort
@ 0x157860d _ZN9__gnu_cxx27__verbose_terminate_handlerEv.cold
@ 0x4d3dfe6 __cxxabiv1::__terminate()
@ 0x4d3e051 std::terminate()
@ 0x4d3e1a5 __cxa_throw
@ 0x157a5e0 std::__throw_system_error()
terminate called recursively
@ 0x179a45b starrocks::TabletSchemaMap::emplace()
@ 0x177fe2e starrocks::TabletMeta::init_from_pb()
@ 0x1781b0f starrocks::TabletMeta::deserialize()
@ 0x1770e2b starrocks::TabletManager::load_tablet_from_meta()
@ 0x17517b8 ZNSt17_Function_handlerIFbllRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEZN9starrocks7DataDir4loadEvEUlliS7_E0_E9_M_invokeERKSt9_Any_dataOlSG_S7
@ 0x1783a8c ZNSt17_Function_handlerIFbSt17basic_string_viewIcSt11char_traitsIcEES3_EZN9starrocks17TabletMetaManager16traverse_headersEPNS5_7KVStoreERKSt8functionIFbllRKNSt7__cxx1112basic_stringIcS2_SaIcEEEEEEUlS3_S3_E_E9_M_invokeERKSt9_Any_dataOS3_SP
@ 0x184fe03 starrocks::KVStore::iterate()
@ 0x1785cad starrocks::TabletMetaManager::traverse_headers()
@ 0x1752f73 starrocks::DataDir::load()
@ 0x1738789 _ZNSt6thread11_State_implINS_8_InvokerISt5tupleIJZN9starrocks13StorageEngine14load_data_dirsERKSt6vectorIPNS3_7DataDirESaIS7_EEEUlvE_EEEEE6_M_runEv
@ 0x4db83a0 execute_native_thread_routine
@ 0x7f216c2a5ea5 start_thread
@ 0x7f216b6aa98d __clone
@ 0x0 (unknown)

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### Bug Description
In order to reduce memory usage, we support shared TabletSchema in version 2.0.x and we add `id` into `TabletSchemaPB`.  
```
message TabletSchemaPB {
    optional KeysType keys_type = 1;    // OLAPHeaderMessage.keys_type
    repeated ColumnPB column = 2;   // OLAPHeaderMessage.column
    optional int32 num_short_key_columns = 3;   // OLAPHeaderMessage.num_short_key_fields
    optional int32 num_rows_per_row_block = 4;  // OLAPHeaderMessage.num_rows_per_data_block
    optional CompressKind compress_kind = 5; // OLAPHeaderMessage.compress_kind
    optional double bf_fpp = 6; // OLAPHeaderMessage.bf_fpp
    optional uint32 next_column_unique_id = 7; // OLAPHeaderMessage.next_column_unique_id
    optional bool DEPRECATED_is_in_memory = 8 [default=false];
    optional int64 id = 9;
}
```
It is conflict with apache-doris
```
message TabletSchemaPB {
    optional KeysType keys_type = 1;    // OLAPHeaderMessage.keys_type
    repeated ColumnPB column = 2;   // OLAPHeaderMessage.column
    optional int32 num_short_key_columns = 3;   // OLAPHeaderMessage.num_short_key_fields
    optional int32 num_rows_per_row_block = 4;  // OLAPHeaderMessage.num_rows_per_data_block
    optional CompressKind compress_kind = 5; // OLAPHeaderMessage.compress_kind
    optional double bf_fpp = 6; // OLAPHeaderMessage.bf_fpp
    optional uint32 next_column_unique_id = 7; // OLAPHeaderMessage.next_column_unique_id
    optional bool is_in_memory = 8 [default=false];
    optional int32 delete_sign_idx = 9 [default = -1];

}
```

`delete_sign_idx` in apache-doris is confilct with `id` in starrocks. The default value of `delete_sign_idx`(-1) is invalid in starrocks, so the upgrade may be failed.

### PR DESCRIPTION
**Pull request by sevev**
This is a bug related to apache-doris compatibility

## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5309 

## Problem Summary(Required) ：

If we create a cluster with apache-doris-0.13, we can upgrade it to starrocks-1.19.x. But when we upgrade it from starrocks-1.19.x to starrocks-2.0.x, be may be failed.

The reason is we add `id` field to `TabletSchemPB` in starrocks-2.0 to support Shared-Schema which conflicts with apache-doris's `TabletSchemaPB`

`TabletSchemaPB` of starrocks-2.0
```
message TabletSchemaPB {
    optional KeysType keys_type = 1;    // OLAPHeaderMessage.keys_type
    repeated ColumnPB column = 2;   // OLAPHeaderMessage.column
    optional int32 num_short_key_columns = 3;   // OLAPHeaderMessage.num_short_key_fields
    optional int32 num_rows_per_row_block = 4;  // OLAPHeaderMessage.num_rows_per_data_block
    optional CompressKind compress_kind = 5; // OLAPHeaderMessage.compress_kind
    optional double bf_fpp = 6; // OLAPHeaderMessage.bf_fpp
    optional uint32 next_column_unique_id = 7; // OLAPHeaderMessage.next_column_unique_id
    optional bool DEPRECATED_is_in_memory = 8 [default=false];
    optional int64 id = 9;
}
```

`TabletSchemaPB` of apache-doris-0.13
```
message TabletSchemaPB {
    optional KeysType keys_type = 1;    // OLAPHeaderMessage.keys_type
    repeated ColumnPB column = 2;   // OLAPHeaderMessage.column
    optional int32 num_short_key_columns = 3;   // OLAPHeaderMessage.num_short_key_fields
    optional int32 num_rows_per_row_block = 4;  // OLAPHeaderMessage.num_rows_per_data_block
    optional CompressKind compress_kind = 5; // OLAPHeaderMessage.compress_kind
    optional double bf_fpp = 6; // OLAPHeaderMessage.bf_fpp
    optional uint32 next_column_unique_id = 7; // OLAPHeaderMessage.next_column_unique_id
    optional bool is_in_memory = 8 [default=false];
    optional int32 delete_sign_idx = 9 [default = -1];

}
```

When we upgrade to starrocks-2.0.x, `delete_sign_idx` will be considered as `id` and we can't guarantee that it's value is meaningful.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by sevev** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: gensrc/proto/olap_file.proto [hunk 1]
```diff
@@ -206,7 +206,8 @@ message TabletSchemaPB {
     optional double bf_fpp = 6; // OLAPHeaderMessage.bf_fpp
     optional uint32 next_column_unique_id = 7; // OLAPHeaderMessage.next_column_unique_id
     optional bool DEPRECATED_is_in_memory = 8 [default=false];
-    optional int64 id = 9;
+    optional int64 id = 9; // deprecated
```
Comment (sduzh) [inline gensrc/proto/olap_file.proto:209]:
```suggestion
    optional int64 deprecated_id = 9; // deprecated
```

File: gensrc/proto/olap_file.proto [hunk 2]
```diff
@@ -206,7 +206,8 @@ message TabletSchemaPB {
...
+    optional int64 schema_id = 50;
```
Comment (sduzh) [inline gensrc/proto/olap_file.proto:210]:
```suggestion
    optional int64 id = 50;
```

Comment (sevev) [reply gensrc/proto/olap_file.proto]: > 

done, change in https://github.com/StarRocks/starrocks/pull/5314/commits/6a81340bbfff836c866b68f805fb80b5268e3e2c

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by sduzh — APPROVED** [review]

**Comment by imay** [comment]
@sevev If you change the code like that, how can starrocks be compatible with itself? Old version store id in position 9, but new version can't get the right id. What will it cause?

**Comment by sevev** [comment]
> @sevev If you change the code like that, how can starrocks be compatible with itself? Old version store id in position 9, but new version can't get the right id. What will it cause?

The old id stored in position 9 is deprecated, new version starrocks will start without `id`. The new version of Be will construct tablet_schema when it starts, just like upgrading from starrocks-1.19.x to starrocks-2.0.x. The Shared-Schema will be rebuilt during the subsequent execution

**Comment by imay** [comment]
> > @sevev If you change the code like that, how can starrocks be compatible with itself? Old version store id in position 9, but new version can't get the right id. What will it cause?
> 
> The old id stored in position 9 is deprecated, new version starrocks will start without `id`. The new version of Be will construct tablet_schema when it starts, just like upgrading from starrocks-1.19.x to starrocks-2.0.x. The Shared-Schema will be rebuilt during the subsequent execution

Thanks, I got you.

**Review by imay — APPROVED** [review]

**Pull request merged**